File: week07/count_word.pl
#!/usr/bin/perl
use strict;
use warnings;

use English;

require "total_words.pl";

sub getWordCounts {
    my ($input) = @ARG;

    my %result = ("_total" => 0);
    foreach my $word (findWords($input)) {
        $word = lc $word;
        if (!$result{$word}) {
            $result{$word} = 0;
        }

        ++$result{$word};
        ++$result{"_total"};
    }

    return %result;
}

if (!caller) {
    my $word = lc $ARGV[0];
    my %wordCounts = getWordCounts(\*STDIN);
    printf("%s occurred %d times\n", $word, $wordCounts{$word} || 0);
}

1;

File: week07/count_word.py
#!/usr/bin/python3

import io, sys
from total_words import findWords

__all__ = ["getWordCounts"]

def getWordCounts(input):
    result = {"_total": 0}
    for word in findWords(input):
        word = word.lower()
        if not word in result:
            result[word] = 0

        result[word] += 1
        result["_total"] += 1

    return result

if __name__ == "__main__":
    word = sys.argv[1].lower()
    wordCounts = getWordCounts(io.TextIOWrapper(sys.stdin.buffer, encoding = "utf-8"))
    print("{} occurred {} times".format(word, wordCounts.get(word, 0)))
